internal/hvsocket/hvsocket.go

```diff
@@ -0,0 +1,81 @@
+//go:build windows
+// +build windows
+
+package hvsocket
+
+import (
+	"context"
+	"fmt"
+	"unsafe"
+
+	"github.com/Microsoft/go-winio/pkg/guid"
+	"golang.org/x/sys/windows"
+
+	"github.com/Microsoft/hcsshim/internal/resources"
+)
+
+const (
+	addressFlagPassthru            = 0x00000001
+	ioCtlHVSocketUpdateAddressInfo = 0x21c004
+)
+
+type addressInfo struct {
+	systemID         guid.GUID
+	virtualMachineID guid.GUID
+	siloID           guid.GUID
+	flags            uint32
+}
+
+type addressInfoCloser struct {
+	handle windows.Handle
+}
+
+var _ resources.ResourceCloser = addressInfoCloser{}
+
+func (aic addressInfoCloser) Release(_ context.Context) error {
+	return windows.CloseHandle(aic.handle)
+}
+
+func CreateAddressInfo(cid, vmid guid.GUID, passthru bool) (resources.ResourceCloser, error) {
```

Comment:
Do we anticipate cases where the `passthru` value will be false? If not, can we make this an implementation detail of this function and always pass the flag?

Reply:
it can be relevant for Krypton... but we can revise this if necessary.

Reply:
I've did some refactoring and added comments. lmk your thoughts.